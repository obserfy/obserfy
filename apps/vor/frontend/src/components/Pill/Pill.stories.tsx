import { FC } from "react"
import Pill from "./Pill"

export default {
  title: "Core/Pill",
  component: Pill,
  parameters: {
    componentSubtitle: "Just a simple Pill",
  },
}

export const Basic: FC = () => <Pill text="You" />
