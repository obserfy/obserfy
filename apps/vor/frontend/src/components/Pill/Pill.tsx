import { Trans } from "@lingui/macro"
import { FC } from "react"
import { Box, BoxProps, Text } from "theme-ui"
import { borderFull } from "../../border"

interface Props extends Omit<BoxProps, "css"> {
  text?: string
}
export const Pill: FC<Props> = ({ sx, color, text, ...props }) => (
  <Box
    py={1}
    px={2}
    sx={{
      ...borderFull,
      borderRadius: "circle",
      flexShrink: 0,
      display: "inline-block",
      ...sx,
    }}
    {...props}
  >
    <Text
      color={color}
      sx={{
        display: "block",
        fontSize: 0,
        textTransform: "capitalize",
      }}
    >
      <Trans id={text} />
    </Text>
  </Box>
)

export default Pill
