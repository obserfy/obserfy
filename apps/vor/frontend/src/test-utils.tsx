/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
