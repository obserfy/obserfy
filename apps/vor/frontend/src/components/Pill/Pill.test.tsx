import { render } from "../../test-utils"
import { Basic } from "./Pill.stories"

describe("Pill", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
