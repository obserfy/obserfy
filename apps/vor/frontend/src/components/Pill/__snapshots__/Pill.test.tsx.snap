// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 8px;
  padding-right: 8px;
  border-width: 1px;
  border-style: solid;
  border-color: var(--theme-ui-colors-border);
  border-radius: 99999px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: inline-block;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 14px;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  display: block;
  font-size: 13px;
  text-transform: capitalize;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span
        class="emotion-2"
      >
        You
      </span>
    </div>
  </div>
</div>
`;
